Proposal Team PixelTalk

Patrick-Houlihan opened the pull request.

Patrick-Houlihan committed:
Add files via upload

Patrick-Houlihan committed:
Update proposal.md

Patrick-Houlihan commented:
Team 16 by the way

charlieroberts commented:
OK, cool. Some questions:

1. You say you'll use WebGL for drawing... is that necessary? Seems like a 2D canvas context would make more sense, although I suppose you could do some interesting manipulation with shaders if you were drawing to a WebGL texture. If you do decide to go the WebGL route you'll need to use some type of library to make things easier on yourselves... I like [regl](https://github.com/regl-project/regl). But, really, you should use a 2D canvas context unless you have a really good reason not to.

2. How will you push the realtime communication notifications? Via a WebSocket connection? Or by periodically querying the database?

Otherwise this seems like a solid project that will entail some fairly intense UI work. Looking forward to seeing it in action!

Patrick-Houlihan commented:
> 
> 
> OK, cool. Some questions:
> 
>     1. You say you'll use WebGL for drawing... is that necessary? Seems like a 2D canvas context would make more sense, although I suppose you could do some interesting manipulation with shaders if you were drawing to a WebGL texture. If you do decide to go the WebGL route you'll need to use some type of library to make things easier on yourselves... I like [regl](https://github.com/regl-project/regl). But, really, you should use a 2D canvas context unless you have a really good reason not to.
> 
>     2. How will you push the realtime communication notifications? Via a WebSocket connection? Or by periodically querying the database?
> 
> 
> Otherwise this seems like a solid project that will entail some fairly intense UI work. Looking forward to seeing it in action!

Definitely switching to canvas, I put WebGL there are an example of something I have experience in, and forgot to switch it to canvas, which we did discuss during a team meeting. Two js is another we were considering. Sorry for the inaccurate detail.

Commit:
Switch to correct Repo

Commit:
Switched to correct Repo

noahvolson committed:
Added basic .gitignore

noahvolson committed:
Merge branch 'master' into Server_Noah

noahvolson committed:
Preliminary database connection and functions

noahvolson committed:
Added Github OAuth groundwork

Patrick-Houlihan committed:
basic drawing canvas done

Patrick-Houlihan committed:
updated drawing stuff

Commit:
Drawing Updates

Commit:
Update README.md

Commit:
Add files via upload

Commit:
Delete draw_view.png

Commit:
Drawing Updates

Commit:
Delete draw_view.PNG

Commit:
Drawing Updates

Commit:
Merge remote-tracking branch 'origin/UI_Jake' into UI_Jake

# Conflicts:
#	public/images/draw_view.png

Commit:
Convert canvas to Data URL

noahvolson committed:
Merge branch 'drawing' into Server_Noah

noahvolson committed:
Can logout via the /logout route

noahvolson committed:
Functionality to create account when one doesn't exist

Patrick-Houlihan committed:
Merge pull request #1 from WebwareA42020/Server_Noah

Server noah

Commit:
Updated drawing code

Commit:
Updated drawing code

Commit:
Updated drawing code

Patrick-Houlihan committed:
commit this

noahvolson committed:
Started transition from handlebars to HTML

carlypereira committed:
Updates to Gallery Page and GithubOauth

I changed some of the gallery page html, and began implementing the GitHub Oauth.

carlypereira committed:
Login Page

Basic login page

carlypereira committed:
gallery nav bar changes

noahvolson committed:
Copied Carly's login page + css into project

carlypereira committed:
Login page style changes

noahvolson committed:
Tweaked user data format

noahvolson committed:
Added route for accessing friend data

noahvolson committed:
Display friend data on home page

noahvolson committed:
Added userdata back to page

noahvolson committed:
Added basic logout button

Patrick-Houlihan committed:
removed deprecated code

noahvolson committed:
Added Jakes CSS file

Patrick-Houlihan committed:
Pulling Noah's updated server

# Conflicts:
#	app.js
#	package-lock.json
#	package.json
#	public/css/main.css
#	public/js/draw.js

Commit:
Updates

noahvolson committed:
Removed handlebars (again?)

noahvolson committed:
Merge branch 'Jake' into Server_Noah2

carlypereira committed:
Updating index.html to match gallery.handlebars

carlypereira committed:
Add no-wrap to sign in button

noahvolson committed:
Redirect to gallery on profile picture click

carlypereira committed:
README updates

carlypereira committed:
Add logout button back

Patrick-Houlihan committed:
implementing loadImage

Commit:
Updates, js, html, css

Commit:
Fixed Nav spacing

Commit:
Update README.md

noahvolson committed:
Pulled drawing data from the db

noahvolson committed:
Tweaked query to match both sender and receiver fields

noahvolson committed:
function to send drawing data to server

Patrick-Houlihan committed:
getting there on animation

Commit:
CSS Updates

Patrick-Houlihan committed:
Merge pull request #6 from WebwareA42020/UI_Carly

Ui carly

Patrick-Houlihan committed:
removed dependency on env file

Patrick-Houlihan committed:
removed dependency on env file

Patrick-Houlihan committed:
Merge pull request #7 from WebwareA42020/testBranch

Test branch

Commit:
Added ip info

Commit:
Ipinfo updates

noahvolson committed:
merged with master

noahvolson committed:
merged with Jake

noahvolson committed:
Fixed broken timer

Patrick-Houlihan committed:
professor roberts is my hero

Patrick-Houlihan committed:
removed traces of testing

noahvolson committed:
Pull title and receiver from inputs

Patrick-Houlihan committed:
remove console clutter

Patrick-Houlihan committed:
Merge branch 'master' into animateImage

Patrick-Houlihan committed:
Merge pull request #8 from WebwareA42020/animateImage

Animate image

noahvolson committed:
Drawn images placed in database

noahvolson committed:
Show all messages to you in inbox

Commit:
Added basic web socket messaging, storing users in array with their username and connection object.

csmbrad committed:
More with sockets, outlining what needs to be done

Patrick-Houlihan committed:
Made profile pic changes work on homepage

csmbrad committed:
Socket Work, Sockets Broken.

Patrick-Houlihan committed:
Merge branch 'master' into profilePic2

Patrick-Houlihan committed:
Merge pull request #9 from WebwareA42020/profilePic2

Profile pic2

Patrick-Houlihan committed:
fixed navbar bug

noahvolson committed:
Sockets actually work now (pog)

Patrick-Houlihan committed:
added readme again

noahvolson committed:
Working notifications

noahvolson committed:
Tweaked notification button

noahvolson committed:
Fixed merge conflicts with master
